### PR TITLE
Add support for git stash

### PR DESCRIPTION
**Pull request by chetan**
This adds support for browsing `git stash` and dropping selected items.

### CODE REVIEW

File: lib/modules/helpers/stash.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+gf_helper_stash_menu_content() {
+  printf "%s" "$GRAY" "$BOLD" '$ ' "$CYAN" "$BOLD"
+  # shellcheck disable=2046
+  printf "%s " "git stash list"
+  if [ -n "$1" ]; then
+    printf "%s" "| grep " "$(quote_params "$@")"
+    printf "%s\n\n" "$NORMAL"
+    git stash list | grep $1
```
Comment (bigH) [inline lib/modules/helpers/stash.sh:10]: quote the `$1` here.

File: lib/modules/helpers/stash.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+gf_helper_stash_menu_content() {
+  printf "%s" "$GRAY" "$BOLD" '$ ' "$CYAN" "$BOLD"
+  # shellcheck disable=2046
```
Comment (bigH) [inline lib/modules/helpers/stash.sh:5]: this can be removed - i imagine it was cargo-culted.

File: lib/modules/helpers/stash.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+gf_helper_stash_preview_content() {
+  if [ -z "$1" ]; then
+    echo "nothing to show"
+  else
+    STASH_ID=$(echo "$1" | cut -d':' -f1)
+
+    # shellcheck disable=2086
+    git stash show -p $STASH_ID | gf_diff_renderer
```
Comment (bigH) [inline lib/modules/helpers/stash.sh:24]: quote the `$STASH_ID`

File: lib/modules/helpers/stash.sh [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    printf "%s\n\n" "$NORMAL"
+    git stash list
+  fi
+}
+
+gf_helper_stash_preview_content() {
+  if [ -z "$1" ]; then
+    echo "nothing to show"
+  else
+    STASH_ID=$(echo "$1" | cut -d':' -f1)
```
Comment (bigH) [inline lib/modules/helpers/stash.sh:21]: quote the `$(...)`

File: lib/modules/helpers/stash.sh [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+    # shellcheck disable=2086
```
Comment (bigH) [inline lib/modules/helpers/stash.sh:23]: you should be able to remove this if you quote the below.

File: lib/modules/helpers/stash.sh [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    git stash show -p $STASH_ID | gf_diff_renderer
+  fi
+}
+
+gf_helper_stash_drop() {
+  if [ -z "$1" ]; then
+    echo "nothing to show"
+  else
+    STASH_ID=$(echo "$1" | cut -d':' -f1)
```
Comment (bigH) [inline lib/modules/helpers/stash.sh:32]: same as above function

File: lib/modules/helpers/stash.sh [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+    # shellcheck disable=2086
```
Comment (bigH) [inline lib/modules/helpers/stash.sh:34]: same as above function

File: lib/modules/helpers/stash.sh [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    git stash drop $STASH_ID
```
Comment (bigH) [inline lib/modules/helpers/stash.sh:35]: same as above function

File: lib/modules/stash.sh [hunk 9]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+GIT_FUZZY_DROP_KEY=${GIT_FUZZY_DROP_KEY:-Alt-D}
+
+# shellcheck disable=2016
+GF_STASH_HEADER='
+'"${WHITE}Enter${NORMAL} to ${GREEN}QUIT${NORMAL}"'
+
+    '"${YELLOW}${BOLD}∆${NORMAL} ${GREEN}drop${NORMAL}  ${WHITE}$GIT_FUZZY_DROP_KEY${NORMAL}     ${GRAY}-- drop the selected stash${NORMAL}"'
```
Comment (bigH) [inline lib/modules/stash.sh:9]: not a blocker: it'd be nice to have an `apply` and `pop` here.

**Comment by bigH** [comment]
this PR is a great idea. tried it locally and the impl is 👌 

you simply need to run [`shellcheck`](https://github.com/koalaman/shellcheck) on the code changes
 - you have some disables that are not needed
 - you have some things that don't pass lint

I realize that I should set up a github action to verify shellcheck passes so folks know and I don't have to do it manually

**Comment by bigH** [comment]
made another pr, merged it with fixes.

**Comment by chetan** [comment]
This initial PR was just a float for some feedback, as the initial work was a quick hack and [obviously] a lot of blind copy/paste to get things working. It wasn't totally clear how to integrate a new command but it didn't take long to get things work either. 

Thanks for cleaning up and merging!

**Pull request merged**